iothub/device/src/Utilities/ObjectConversionHelper.cs

```diff
@@ -20,7 +20,7 @@ internal static bool TryCastOrConvert<T>(object objectToCastOrConvert, PayloadCo
             {
                 // We'll serialize the object back to JSON using the user's configured payload convention
                 // and then to the type of their choosing with the same payload convention.
-                value = payloadConvention.GetObject<T>(payloadConvention.GetObjectBytes(objectToCastOrConvert));
+                value = payloadConvention.GetObject<T>(DefaultPayloadConvention.Instance.GetObjectBytes(objectToCastOrConvert));
```

Comment:
The comment states that both of these operations are expected to use the user's payload convention.
Your change is updating one of the conversions to always use Newtonsoft.

Can you elaborate on this change?

Reply:
The comment needs updating.

The twin was deserialized with the default payload converter. So using the user's (when it is STJ) will fail to create object bytes that are compatible. Presumably the object is a JObject or something, right?

So we need to use the same payload convention used to deserialize the twin to serialize it back, and then use the user's one to deserialize.

Reply:
Updated the comment